backend-api/src/functions/asyncActiveSession/tokenService/tokenService.ts

```diff
@@ -102,6 +106,49 @@ export class TokenService implements ITokenService {
 
     return successResult(jwks);
   }
+
+  private async decryptJwe(
+    key: Uint8Array,
+    iv: Uint8Array,
+    ciphertext: Uint8Array,
+    tag: Uint8Array,
+    additionalData: Uint8Array,
+  ): Promise<Result<Uint8Array>> {
+    const webcrypto = crypto.webcrypto as unknown as Crypto;
+
+    let cek: CryptoKey;
+    try {
+      cek = await webcrypto.subtle.importKey("raw", key, "AES-GCM", false, [
+        "decrypt",
+      ]);
+    } catch (error) {
+      return errorResult({
+        errorMessage: `Error converting cek to CryptoKey. ${error}`,
+        errorCategory: "SERVER_ERROR",
+      });
+    }
+
+    let decryptedBuffer: ArrayBuffer;
+    try {
+      decryptedBuffer = await webcrypto.subtle.decrypt(
+        {
+          name: "AES-GCM",
+          iv,
+          additionalData,
+          tagLength: 128,
+        },
+        cek,
+        Buffer.concat([new Uint8Array(ciphertext), new Uint8Array(tag)]),
+      );
+    } catch (error) {
+      return errorResult({
+        errorMessage: `Error decrypting JWE. ${error}`,
+        errorCategory: "SERVER_ERROR",
+      });
+    }
+
+    return successResult(new Uint8Array(decryptedBuffer));
```

Comment:
Feedback from Ana recommends to return this decoded as well. Have tested already, will commit shortly

Reply:
Done